pkg/ddevapp/providerPantheon.go

```diff
@@ -0,0 +1,365 @@
+		err = util.DownloadFile(destFile, backup.DownloadURL, true)
+		if err != nil {
+			return "", "", err
+		}
+	}
+
+	if backupType == "files" {
+		importPath = fmt.Sprintf("files_%s", p.Environment)
+	}
+
+	return destFile, importPath, nil
+}
+
+// PostImport provides hooks which should happen after an import on a site which uses this provider. In pantheons case, we need to write a provider.env file to provide pressflow with the appropriate environment vars.
+func (p *PantheonProvider) PostImport(url string) error {
+	var doc bytes.Buffer
+	var err error
+	f, err := os.Create(p.config.GetPath("provider.env"))
+	if err != nil {
+		return err
+	}
+	defer util.CheckClose(f)
+	templ := template.New("environment template")
+	templ, err = templ.Parse(PantheonEnvironmentTemplate)
+	if err != nil {
+		return err
+	}
+	templateVars := map[string]string{
+		"framework": p.config.AppType,
+		// It's not great to have to use this as parameter to this function rather than being
+		// able to askt the config for it, but as long as config and platform are
+		// separate we're left with this.
+		"base_url": url,
+	}
+
+	err = templ.Execute(&doc, templateVars)
+	if err != nil {
+		return err
+	}
+	if err != nil {
+		return err
+	}
+	_, err = f.WriteString(doc.String())
+
+	return err
+}
+
+// prepDownloadDir ensures the local download cache directories are created and writeable.
+func (p *PantheonProvider) prepDownloadDir() {
+	destDir := p.getDownloadDir()
+	err := os.MkdirAll(destDir, 0755)
+	util.CheckErr(err)
+}
+
+func (p *PantheonProvider) getDownloadDir() string {
+	userDir, err := homedir.Dir()
+	util.CheckErr(err)
+	destDir := filepath.Join(userDir, ".ddev", "pantheon", p.config.Name)
+	return destDir
+}
+
+// getPantheonBackupLink will return a URL for the most recent backyp of archiveType that exist with the BackupList specified.
+func getPantheonBackupLink(archiveType string, bl *pantheon.BackupList, session *pantheon.AuthSession) (*pantheon.Backup, error) {
+	latestBackup := pantheon.Backup{}
+	for i, backup := range bl.Backups {
+		if backup.ArchiveType == archiveType && backup.Timestamp > latestBackup.Timestamp {
+			latestBackup = bl.Backups[i]
+		}
+	}
+
+	if latestBackup.Timestamp != 0 {
+		// Get a time-limited backup URL from Pantheon. This requires a POST of the backup type to their API.
+		err := session.Request("POST", &latestBackup)
+		if err != nil {
+			return &pantheon.Backup{}, fmt.Errorf("could not get backup URL: %v", err)
+		}
+
+		return &latestBackup, nil
+	}
+
+	// If no matches were found, just return an empty backup along with an error.
+	return &pantheon.Backup{}, fmt.Errorf("could not find a backup of type %s", archiveType)
+}
+
+// environmentPrompt contains the user prompts for interactive configuration of the pantheon environment.
+func (p *PantheonProvider) environmentPrompt() error {
+	_, err := p.GetEnvironments()
+	if err != nil {
+		return err
+	}
+
+	if p.Environment == "" {
+		p.Environment = "dev"
```

Comment:
"live" seems like the most appropriate default, but i suppose with pantheon that doesn't exist until a site is close to done (and a credit card has been provided).